include/dali/core/tensor_layout.h

```diff
@@ -193,6 +193,15 @@ class TensorLayout {
   DALI_HOST_DEV
   constexpr bool empty() const noexcept { return size() == 0; }
 
+  void resize(size_t new_size, char value = '?') noexcept {
```

Comment:
Nitpick: I'd like a more meaningful name
```suggestion
  void resize(size_t new_size, char fill_value = '?') noexcept {
```